### DISCUSSION TITLE
how to use `-Z shell-escape-cwd` with V2 `-X build`

### DISCUSSION THREAD

**Discussion by goyalyashpal** [tectonic-typesetting/tectonic#1113] [Q&A]
>  This works with the V2 CLI as expected btw.
>
> One side effect though is that intermediate files will be placed in the directory thats marked as working-directory, depending on the commands that were used. I dont know how that could be solved generally without potentially problematic approaches.
>
> \- _Originally posted by @9SMTM6 in https://github.com/tectonic-typesetting/tectonic/issues/909#issuecomment-1295172847_

Q:
- how to use `-Z shell-escape-cwd` with V2 `-X` ?
- and how to specify that via `Tectonic.toml` ?

I tried doing the following: 

```sh-session
$ tectonic -V
Tectonic 0.14.1

$ tectonic -Zhelp | grep -EniA1 shell-escape
10:    -Z shell-escape             Enable \write18
11:    -Z shell-escape-cwd         Working directory to use for \write18. Use $(pwd) for same behaviour as
12-                                    most other engines (e.g. for relative paths in \inputminted).
13:                                    Implies -Z shell-escape
14-    -Z deterministic-mode       Force a deterministic build environment. Note that setting

$ tectonic -X build -Z shell-escape-cwd
error: Found argument '-Z' which wasn't expected, or isn't valid in this context

USAGE:
    build [FLAGS] [OPTIONS]

For more information try --help

$ tectonic -Z shell-escape-cwd -X build
error: Invalid value for '-Z <option>...': '-Z shell-escape-cwd=<path>' requires a value but none was supplied

```

**Comment by eljamm** [comment]
- With `-X`:
You can pass the `-Z shell-escape-cwd` to the [compile](https://tectonic-typesetting.github.io/book/latest/v2cli/compile.html) command:

```
tectonic -X compile -Z shell-escape-cwd=<path> myfile.tex
```

- With the [Tectonic.toml](https://tectonic-typesetting.github.io/book/latest/ref/tectonic-toml.html#the-tectonictoml-file) file:

```toml
[[output]]
name = "default"
type = "pdf"
shell_escape_cwd = "<path>"
```

The `"<path>"` directory can be either absolute or relative to the root file, but it must exist or the compilation will fail.
